man/fi_poll.3.md

```diff
@@ -195,6 +204,62 @@ being closed, otherwise the call will return -FI_EBUSY.
 Waits on a wait set until one or more of its underlying wait objects
 is signaled.
 
+## fi_trywait
+
+The fi_trywait() call is used in conjunction with native operating
+system calls to block on wait objects, such as file descriptors.  It
+must be called, with a successful return value of FI_SUCCESS, prior to
+blocking on a native wait object, or the wait object may not be
+signaled, resulting in an application hang.  The following
+psuedo-code demonstrates the use of fi_trywait in conjunction with
+the OS select call.
+
+```c
+fi_control(&cq->fid, FI_GETWAIT, (void *) &fd);
+FD_ZERO(&fds);
+FD_SET(fd, &fds);
+
+while (1) {
+	if (fi_trywait(&cq, 1) == 0)
+		select(fd + 1, &fds, NULL, &fds, &timeout);
+
+	do {
+		ret = fi_cq_read(cq, &comp, 1);
+	} while (ret > 0);
+}
+```
+
+Fi_trywait() will return FI_SUCCESS if it is safe to block on the wait object(s)
+corresponding to the fabric descriptor(s), or -FI_EAGAIN if blocking
+could hang the application.  For example, -FI_EAGAIN may indicate that
+there are completions available on a CQ.
+
+The call takes an array of fabric descriptors.  For each wait object
+that will be passed to the native wait routine, the corresponding
+fabric descriptor should first be passed to fi_trywait.  All fabric
+descriptors passed into a single fi_trywait call must make use of the
+same underlying wait object.
```

Comment:
"All fabric descriptors passed into a single fi_trywait call must make use of the same underlying wait object."

I think this is a strange restriction, since for non-wait-set wait objects, there's no way for an application to request that, e.g., two CQs use the same wait object under the hood, right?  And for wait sets we would be force to pass at most a single wait set by the same restriction.  So I think the above restriction implies that the `count` argument to `fi_trywait` will always be `1`.